torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -156,7 +156,127 @@ def generate_ttir(kernel, kwargs):
     ttir_module = src.make_ir(options, context)
     if not ttir_module.verify():
         raise Exception("Verification for TTIR module has failed")
-    return str(ttir_module), ordered_tensor_names
+
+    return ttir_module, ordered_tensor_names
+
+
+def ttir_to_functions(ttir_module):
+    """
+    Walk the `ttir_module` bottom up to mine the `functions` from
+    the structured MLIR entities representing the Triton kernel
+    (mlir::Operation, mlir::Block, mlir::Region).
+    """
+    functions = {}
+
+    # block id --> op result (Intermediate) --> one or more ops
+    op_stack: Dict[int, Dict[Intermediate, List[Op]]] = defaultdict(
+        lambda: defaultdict(list)
+    )
+    region_id_to_block_ids: Dict[int, List[int]] = defaultdict(list)
+    block_id_to_block_arg_ids: Dict[int, List[int]] = {}
+    next_fake_intermediate = 0
+
+    def mlir_to_functions(op):
```

Comment:
Lets add a type to this op to make it clear

Reply:
I can add some more type hints, but for exposed bindings this is not trivial (as their types are not visible to mypy), so I'd prefer skipping type hints for `op` and `ttir_module`.

Reply:
Would it make sense to add a typehint file for mypy? Similar to our own C++ bindings. Just a suggestion, not a blocker